### PR TITLE
Add conditional to override cycles checking if is_reverse_complement

### PR DESCRIPTION
**Pull request by diitaz93**
## Description

The OverrideCycles and BarcodeMismatches2 attributes of a FlowCellSample was set incorrectly for samples requiring reverse complement. This PR adds the fix for reverse complement samples.

### Added

- SampleSheetCreator property `is_reverse_complement` instead of making it an internal variable
- Tests

### Changed

- 

### Fixed

- The building of index2 in `add_override_cycles` now checks if the sample sheet `is_reverse_complement` to invert order of terms "I" and "N"
- `get_hamming_distance_index_2` receives `is_reverse_complement` to take into account for calculation

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b fix-override-cycles -a
    ```

### How to test

- [ ] Generate a sample sheet for a NovaSeqX
- [ ] Generate a sample sheet for a NovaSeq6000 that requires reverse complement of indexes

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/apps/demultiplex/sample_sheet/index.py [hunk 1]
```diff
@@ -156,19 +157,29 @@ def get_hamming_distance_index_1(sequence_1: str, sequence_2: str) -> int:
...
-    return get_hamming_distance(
-        str_1=sequence_1[-shortest_index_length:], str_2=sequence_2[-shortest_index_length:]
+    return (
+        get_hamming_distance(
+            str_1=sequence_1[:shortest_index_length], str_2=sequence_2[:shortest_index_length]
+        )
+        if is_reverse_complement
+        else get_hamming_distance(
+            str_1=sequence_1[-shortest_index_length:], str_2=sequence_2[-shortest_index_length:]
+        )
```
Comment (islean) [inline cg/apps/demultiplex/sample_sheet/index.py:175]: Does this work? I interpret this as "start on the negative shortest_index_length and continue until the end", but maybe my Python knowledge is off? I was thinking it should be "length of sequence_1 - shortest_index_length" or something?

**Review by islean — COMMENTED** [review]
Don't know enough to approve but the gist of it looks good!

**Review by Karl-Svard — APPROVED** [review]
Nice! Just modify the testing to use a non NIPT flow cell instead of altering the existing fixture 👍 I think Kalle will give more specifics

File: cg/apps/demultiplex/sample_sheet/index.py (hunk 1 above)
Comment (diitaz93) [reply cg/apps/demultiplex/sample_sheet/index.py:159]: This means exactly length of sequence_1 - shortest_index_length 😊 . The minus means you start counting from the tail to the head.

**Comment by diitaz93** [comment]
## Tests on stage:
Successful sample sheet for NovaSeqX and NovaSeq6000

**Pull request merged**